### PR TITLE
runtimevar: Expose DecoderByName directly

### PR DESCRIPTION
**Pull request by eliben**
For #1782

This should show the actually useful documentation of this function in the godoc, and list its arguments

### CODE REVIEW

File: runtimevar/runtimevar.go [hunk 1]
```diff
@@ -509,9 +509,6 @@ func DecryptDecode(ctx context.Context, k *secrets.Keeper, post Decode) Decode {
 }
 
 // DecoderByName returns a *Decoder based on decoderName.
-// It is intended to be used by VariableURLOpeners in driver packages.
```
Comment (vangent) [inline runtimevar/runtimevar.go]: Let's keep this line in the docstring, to let end users know they shouldn't be using this directly.

**Comment by vangent** [comment]
```
Incompatible changes:
- DecoderByName: changed from var to func
```

File: runtimevar/runtimevar.go (hunk 1 above)
Comment (eliben) [reply runtimevar/runtimevar.go]: Done

**Comment by eliben** [comment]
Thanks @shantuo , @vangent and I chatted about this

**Pull request merged**